src/Data/Food/Ingredient.elm

```diff
@@ -0,0 +1,59 @@
+module Data.Food.Ingredient exposing
+    ( Ingredient
+    , decodeIngredients
+    )
+
+import Data.Food.Process as Process exposing (Process)
+import Dict exposing (Dict)
+import Json.Decode as Decode exposing (Decoder)
+import Json.Decode.Extra as DE
+import Json.Decode.Pipeline as Pipe
+
+
+type alias Ingredient =
+    { name : String
+    , conventional : Process
+    , labels : Labels
+    }
+
+
+type alias Labels =
+    { organic : Maybe Process
+    }
+
+
+decodeIngredients : List Process -> Decoder (List Ingredient)
+decodeIngredients processes =
+    let
+        processesDict =
+            processes
+                |> List.map (\process -> ( Process.codeToString process.code, process ))
+                |> Dict.fromList
+    in
+    Decode.list (decodeIngredient processesDict)
```

Comment:
Nit: I know you don't like it much but this can be written as

```elm
decodeIngredients : List Process -> Decoder (List Ingredient)
decodeIngredients =
    List.map (\process -> ( Process.codeToString process.code, process ))
        >> Dict.fromList
        >> decodeIngredient
        >> Decode.list
```

Up to you

Reply:
I'll rewrite it as a single pipeline (removing the `let in`), and keep pipelines instead of chevrons ;)